### ISSUE TITLE
Addition/Subtraction with default(Money) should not throw InvalidCurrencyException

### ISSUE THREAD

**Issue by ptjhuang** [remyvd/NodaMoney#69]
I think the following should succeed, sort of like https://en.wikipedia.org/wiki/Identity_element

`default(Money) + Money.Euro(100)` should give `Money.Euro(100)`

But it throws an exception because the 'currency' of default(Money) is null.

@remyvd and others, what're your opinions?

**Comment by remyvd** [comment]
Solved differently by ac884eb15d312482771421b684723c15b63e55ed and see comment https://github.com/remyvd/NodaMoney/issues/72#issuecomment-611693158.

default(Money) is an uninitialized money struct. To make it more explicit that no currency is given `default(Money) == Money(0, XXX)` (where XXX is _No Currency_).  Allowing to do calculations with money in different currencies, goes against the nature of this library. This library want you to be clear on the currency that you are using.

Allowing `Money(0, XXX) == Money(0, USD) == Money(0, EUR)` seems like a nice idea, because `Money(0, USD) + Money(10, EUR) = Money(10, EUR)` but this can result in code that sometimes fails instead of always failing. See the following example:
```
var a = new Money(5, USD);
var b1 = new Money(5, USD);
var b2 = new Money(4, USD);
var c1 = new Money(10, EUR);
var c2 = new Money(10, USD);

var result = (a - b1) + c1;  // OK, result is EUR 10
var result = (a - b2) + c1;  // Fails!
var result = (a - b1) + c2;  // OK, but result is now USD10
```

Different currencies are different unit of measurements and the need to be converted first. Saying `0 Liters == 0 Meters` would also be strange(although it is technically nothing...).

Also using `default(money)` is, I think, a code-smell: why wouldn't you specified it explicit? Use `Money?` if you want to make it an option to specify.

It also causes a lot of if-statements in the code too support this, in paths that should have as less lines of code as possible for performance.

**Comment by ptjhuang** [comment]
> Also using default(money) is, I think, a code-smell: why wouldn't you specified it explicit? Use Money? if you want to make it an option to specify.

Fair enough. Agreed. Can this be more explicit though? Any operation with default(Money) should result in an InvalidCurrency?
